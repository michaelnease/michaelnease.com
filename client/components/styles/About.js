import styled from "styled-components";

export const AboutStyled = styled.section`
  height: 80rem;
  position: relative;
  background: url("/static/images/bio/bio.gif") no-repeat center center;
  background-size: cover;
  border-bottom: 10px solid ${props => props.theme.colors.black};
  @media (max-width: 700px) {
    height: 100rem;
  }
`;

export const Text = styled.div`
  width: 60rem;
  p {
    text-align: justify;
    font-weight: 300;
    font-size: 1.7rem;
    margin-bottom: 20px;
  }
  @media (max-width: 700px) {
    padding: 3rem;
  }
`;

export const Rex = styled.img`
  box-shadow: ${props => props.theme.effects.bs};
  border: 1px solid ${props => props.theme.effects.black};
  float: left;
  width: 20rem;
  margin: ${props => props.theme.spacing.margin};
  position: relative;
  top: 10px;
`;
