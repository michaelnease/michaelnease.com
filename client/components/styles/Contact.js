import styled from "styled-components";

export const Form = styled.form`
  display: flex;
  position: relative;
  align-content: center;
  justify-content: center;
  background-color: ${props => props.theme.colors.darkGray};
  border-bottom: 10px solid ${props => props.theme.colors.black};
  padding: 3rem;
`;

export const Inputs = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr;
  width: 80rem;
  grid-gap: 1.5rem;

  input,
  textarea {
    border: none;
    background-color: ${props => props.theme.colors.gray};
    padding-left: 3rem;
    font-size: 1.5rem;
    box-shadow: ${props => props.theme.effects.bs};
  }

  input {
    height: 5rem;
  }

  textarea {
    padding: 3rem;
    grid-column: span 2;
  }
`;
