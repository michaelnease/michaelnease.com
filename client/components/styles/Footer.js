import styled from "styled-components";

export const Foot = styled.footer`
  padding: 3rem;
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin: 0 10rem;
  p {
    color: ${props => props.theme.colors.gray};
  }

  @media (max-width: 700px) {
    padding: 1rem;
    margin: 0;
  }
`;
