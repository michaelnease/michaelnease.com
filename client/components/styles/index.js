export * from "./Header";
export * from "./Quotes";
export * from "./About";
export * from "./Technology";
export * from "./Portfolio";
export * from "./Contact";
export * from "./Footer";
export * from "./Send-Message-Button";

export * from "./Texture";
