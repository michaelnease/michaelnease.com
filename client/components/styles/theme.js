export const theme = {
  colors: {
    black: "#141414",
    white: "#fff",
    gray: "#e0e0e0",
    darkGray: "#a9a9a9"
  },
  spacing: {
    margin: "15px",
    padding: "15px"
  },
  effects: {
    bs: "-5px 5px 5px 0px rgba(0, 0, 0, 0.2)"
  },
  bar: {
    height: "60px"
  }
};
