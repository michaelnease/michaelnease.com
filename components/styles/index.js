export * from "./Header";
export * from "./Quotes";
export * from "./Bio";
export * from "./Technology";
export * from "./Portfolio";

export * from "./Texture";
