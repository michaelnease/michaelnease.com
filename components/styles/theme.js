export const theme = {
  colors: {
    black: "#141414",
    white: "#fff",
    gray: "#e0e0e0"
  },
  spacing: {
    margin: "15px"
  },
  effects: {
    bs: "-5px 5px 5px 0px rgba(0, 0, 0, 0.2)"
  },
  bar: {
    height: "60px"
  }
};
