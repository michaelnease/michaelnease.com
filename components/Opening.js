import { Quotes, Dropdown, Cites, Quote, Texture, Cite } from "./styles";

const Opening = () => {
  return (
    <Quotes>
      <Texture>
        <Dropdown />
        <Cites>
          <ul>
            <li>
              <Quote>
                <q>
                  If debugging is the process of removing software bugs, then
                  programming must be the process of putting them in.
                </q>
                <Cite>Edsger Dijkstra</Cite>
              </Quote>
            </li>
            <li>
              <Quote>
                <q>
                  The best performance improvement is the transition from the
                  nonworking state to the working state.
                </q>
                <Cite>J. Osterhout</Cite>
              </Quote>
            </li>
            <li>
              <Quote>
                <q>
                  Don’t worry if it doesn’t work right. If everything did, you’d
                  be out of a job.
                </q>
                <Cite>
                  Don’t worry if it doesn’t work right. <br />
                  If everything did, you’d be out of a job.
                </Cite>
              </Quote>
            </li>
          </ul>
        </Cites>
      </Texture>
    </Quotes>
  );
};

export default Opening;
