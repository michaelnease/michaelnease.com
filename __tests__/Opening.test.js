import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import { render } from "react-testing-library";
import "jest-styled-components";
import { ThemeProvider } from "styled-components";

import { theme } from "../components/styles/theme";
import Opening from "../components/Opening";

configure({ adapter: new Adapter() });

describe("<Opening />", () => {
  const wrapper = shallow(<Opening />);

  it("Should render <Dropdown /> element", () => {
    expect(wrapper.find("Dropdown").length).toBe(1);
  });

  it("Should render <Texture /> element", () => {
    expect(wrapper.find("Texture").length).toBe(1);
  });

  it("Should render 3 <Cites /> elements", () => {
    expect(wrapper.find("Cites ul li").length).toBe(3);
  });

  it("Should render a <Cites /> elements", () => {
    expect(
      wrapper
        .find("Cites ul")
        .childAt(0)
        .text()
    ).toBe(
      "If debugging is the process of removing software bugs, then programming must be the process of putting them in.Edsger Dijkstra"
    );
    expect(
      wrapper
        .find("Cites ul")
        .childAt(1)
        .text()
    ).toBe(
      "The best performance improvement is the transition from the nonworking state to the working state.J. Osterhout"
    );
    expect(
      wrapper
        .find("Cites ul")
        .childAt(2)
        .text()
    ).toBe(
      "Design is a funny word. Some people think design means how it looks. But of course, if you dig deeper, it's really how it works.Steve Jobs"
    );
  });

  it("Should match Snapshot", () => {
    const { asFragment } = render(
      <ThemeProvider theme={theme}>
        <Opening />
      </ThemeProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
