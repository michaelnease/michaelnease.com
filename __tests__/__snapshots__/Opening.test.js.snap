// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Opening /> Should match Snapshot 1`] = `
<DocumentFragment>
  .c0 {
  background-color: gray;
  position: relative;
  height: 100vh;
  background: url("/static/images/quotes/quotes.gif") no-repeat center center;
  background-size: cover;
  min-height: 6.2rem;
  border-bottom: 10px solid #141414;
}

.c3 {
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-65%);
  -ms-transform: translate(-50%,-65%);
  transform: translate(-50%,-65%);
  width: 50rem;
  height: 25rem;
}

.c4 {
  background-color: #fff;
  margin: 0 0 15px 0;
  padding: 15px;
  box-shadow: -5px 5px 5px 0px rgba(0,0,0,0.2);
}

.c4 q {
  font-size: 1.4rem;
}

.c5 {
  display: block;
  font-weight: bold;
  text-align: right;
  font-size: 1.4rem;
  margin-top: 1.5rem;
}

.c2 {
  box-shadow: -5px 5px 5px 0px rgba(0,0,0,0.2);
  position: absolute;
  z-index: 1;
  left: 50%;
  -webkit-transform: translate(-50%);
  -ms-transform: translate(-50%);
  transform: translate(-50%);
  width: 124px;
  height: 210px;
  background: url("/static/images/header/mocmennemcom.svg") no-repeat center center;
  -webkit-animation: gGKbGi 0.5s ease-in 1.5s 1 normal both running;
  animation: gGKbGi 0.5s ease-in 1.5s 1 normal both running;
}

.c1 {
  background-image: url("/static/images/common/texture.png");
  height: 100%;
}

@media (max-width:700px) {
  .c3 {
    width: 100%;
  }
}

<section
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      />
      <div
        class="c3"
      >
        <ul>
          <li>
            <blockquote
              class="c4"
            >
              <q>
                If debugging is the process of removing software bugs, then programming must be the process of putting them in.
              </q>
              <cite
                class="c5"
              >
                Edsger Dijkstra
              </cite>
            </blockquote>
          </li>
          <li>
            <blockquote
              class="c4"
            >
              <q>
                The best performance improvement is the transition from the nonworking state to the working state.
              </q>
              <cite
                class="c5"
              >
                J. Osterhout
              </cite>
            </blockquote>
          </li>
          <li>
            <blockquote
              class="c4"
            >
              <q>
                Design is a funny word. Some people think design means how it looks. But of course, if you dig deeper, it's really how it works.
              </q>
              <cite
                class="c5"
              >
                Steve Jobs
              </cite>
            </blockquote>
          </li>
        </ul>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
